src/examples/source/SmartPeakGUI.cpp

```diff
@@ -704,7 +704,10 @@ int main(int argc, char** argv)
     // feature table
     if (features_table_main_window_->visible_)
     {
-      exceeding_table_size_ = !session_handler_.setFeatureTable(application_handler_.sequenceHandler_, features_table_main_window_->table_data_);
+      bool data_changed = false;
+      exceeding_table_size_ = !session_handler_.setFeatureTable(application_handler_.sequenceHandler_,
```

Comment:
There was no way to use the observer pattern already used other places? and then to do the updateDataModificationState in the widget itself (we want to remove code as much as possible from GUI) It looks good this way, it's just not so consistent with the latest modifications.